tests/models/demultiplexing/test_run_parameters.py

```diff
@@ -1,109 +1,261 @@
+):
+    """Test that getting the index length from a NovaSeqX run parameters file returns an int."""
+    # GIVEN a valid RunParametersNovaSeqX object
+
+    # WHEN getting the index length
 
-def test_index_length_different_length(run_parameters_different_index: Path):
+    # THEN the index length is an int
+    assert isinstance(novaseq_x_run_parameters.index_length, int)
+
+
+def test_index_length_novaseq_6000_different_length(
+    run_parameters_novaseq_6000_different_index_path: Path,
+    run_parameters_novaseq_x_different_index_path: Path,
+):
     """Test that getting the index length from a file with different index cycles fails."""
     # GIVEN a RunParameters object created from a file with different index cycles
-    run_parameters = RunParameters(run_parameters_path=run_parameters_different_index)
+    run_parameters_novaseq_6000 = RunParametersNovaSeq6000(
+        run_parameters_path=run_parameters_novaseq_6000_different_index_path
+    )
 
     # WHEN fetching index length
-    with pytest.raises(FlowCellError) as exc_info:
+    with pytest.raises(RunParametersError) as exc_info:
         # THEN assert that an exception was raised since the index cycles are different
-        run_parameters.index_length
+        run_parameters_novaseq_6000.index_length
+        assert str(exc_info.value) == "Index lengths are not the same!"
+
 
-    assert str(exc_info.value) == "Index lengths are not the same!"
+def test_index_length_novaseq_x_different_length(
+    run_parameters_novaseq_x_different_index_path: Path,
+):
+    """Test that getting the index length from a file with different index cycles fails."""
+    # GIVEN a RunParameters object created from a file with different index cycles
+    run_parameters_novaseq_x = RunParametersNovaSeqX(
+        run_parameters_path=run_parameters_novaseq_x_different_index_path
+    )
+    # WHEN fetching index length
+    with pytest.raises(RunParametersError) as exc_info:
+        # THEN assert that an exception was raised since the index cycles are different
+        run_parameters_novaseq_x.index_length
+        assert str(exc_info.value) == "Index lengths are not the same!"
+
+
+def test_get_cycles_novaseq_6000(novaseq_6000_run_parameters: RunParametersNovaSeq6000):
+    """Test that the read and index cycles are read correctly for NovaSeqX run parameters."""
+    # GIVEN a NovaSeq6000 run parameters object
+
+    # WHEN getting any read cycle
+    read1: int = novaseq_6000_run_parameters.get_read1_cycles()
+    read2: int = novaseq_6000_run_parameters.get_read2_cycles()
+    index1: int = novaseq_6000_run_parameters.get_index1_cycles()
+    index2: int = novaseq_6000_run_parameters.get_index2_cycles()
+
+    # THEN all read cycles are non-negative integers
+    assert isinstance(read1, int)
+    assert read1 >= 0
+    assert isinstance(read2, int)
+    assert read2 >= 0
+    assert isinstance(index1, int)
+    assert index1 >= 0
+    assert isinstance(index2, int)
+    assert index2 >= 0
+
```

Comment:
Would we not want to check that the indices are correct rather than positive? My guess is that there will be more subtle issues that go unnoticed for longer if we get "somewhat" correct numbers.

Reply:
Yes. However, these variables can take a bunch of values which I know are all non-negative integers. Unfortunately, I don't have the knowledge to restrict the testing interval more. All I can say is that if the value is not a non-negative integer I will be 100% sure that the method is not working correctly.